index.bs

```diff
@@ -0,0 +1,505 @@
+     directory failed, this rejects. Creation can fail because there already is a file with the same
+     name, or because the name uses characters that aren't supported in file names on the underlying
+     file system.
+</div></dl>
+
+<div algorithm>
+The <dfn method for=FileSystemDirectoryHandle>getDirectory(|name|, |options|)</dfn> method, when
+invoked, must run these steps:
+
+1. TODO
+
+</div>
+
+### The {{FileSystemDirectoryHandle/getEntries()}} method ### {#api-filesystemdirectoryhandle-getentries}
+
+<div class=note><dl class=domintro>
+  : for await (const |handle| of |directoryHandle| . {{FileSystemDirectoryHandle/getEntries()}}) {}
+  :: Iterates over all entries whose parent is the entry represented by |directoryHandle|.
+</dl></div>
+
+<div algorithm>
+The <dfn method for=FileSystemDirectoryHandle>getEntries()</dfn> method, when invoked, must run
+these steps:
+
+1. TODO
+
+</div>
+
+### The {{FileSystemDirectoryHandle/removeRecursively()}} method ### {#api-filesystemdirectoryhandle-removerecursively}
+
+<div class=note><dl class=domintro>
+  : await |directoryHandle| . {{FileSystemDirectoryHandle/removeRecursively()}}
+  :: Removes the entry represented by |directoryHandle|, and all entries contained within it,
+     recursively.
+</dl></div>
+
+## The {{FileSystemWriter}} interface ## {#api-filesystemwriter}
+
+<xmp class=idl>
+interface FileSystemWriter {
+  Promise<void> write(unsigned long long position, (Blob or ReadableStream) data);
+  Promise<void> truncate(unsigned long long size);
+  Promise<void> close();
+};
+</xmp>
+
+### The {{FileSystemWriter/write()}} method ### {#api-filesystemwriter-write}
+
+<div class=note><dl class=domintro>
+  : await |writer| . {{FileSystemWriter/write()|write}}(|position|, |data|)
+  :: Writes the content of |data| into the file associated with |writer| at position |position|.
+     If |position| is past the end of the file writing will fail and this method rejects.
+</dl></div>
+
+<div algorithm>
+The <dfn method for=FileSystemWriter>write(|position|, |data|)</dfn> method, when invoked, must run
+these steps:
+
+1. TODO
+
+</div>
+
+### The {{FileSystemWriter/truncate()}} method ### {#api-filesystemwriter-truncate}
+
+<div class=note><dl class=domintro>
+  : await |writer| . {{FileSystemWriter/truncate()|truncate}}(|size|)
+  :: Resizes the file associated with |writer| to be |size| bytes long. If |size| is larger than
+     the current file size this pads the file with zero bytes, otherwise it truncates the file.
+</dl></div>
+
+<div algorithm>
+The <dfn method for=FileSystemWriter>truncate(|size|)</dfn> method, when invoked, must run these
+steps:
+
+1. TODO
+
+</div>
+
+# Accessing native filesystem # {#native-filesystem}
+
+## The {{Window/chooseFileSystemEntries()}} method ## {#api-choosefilesystementries}
+
+<xmp class=idl>
+enum ChooseFileSystemEntriesType { "openFile", "saveFile", "openDirectory" };
+
+dictionary ChooseFileSystemEntriesOptionsAccepts {
+  DOMString description;

```

Comment:
Make these `USVString`s, since they're probably going to get converted to UTF-8 at some point.

Reply:
Good point. Done.